### PR TITLE
fix(elasticsearch): handle elasticsearch connection exceptions

### PR DESCRIPTION
**Pull request by k0machi**
In certain cases inability to connect to the elasticsearch can
bring down the entire thread where it is not necessary, this pr
addresses those cases by adding a catch all exception to elasticsearch
calls

[Trello link](https://trello.com/c/anRlwIIL/3366-elasticsearch-updates-shouldnt-fail-tests)

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/results_analyze/test.py [hunk 1]
```diff
@@ -475,11 +476,16 @@ def _get_es_query_from_self(self, data_path_patterns: list):
...
-            filter_path=filter_path,
-            size=10000)
+        try:
+            es_data = ES().search(
+                index=es_index,
+                q=es_query,
+                filter_path=filter_path,
+                size=10000)
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.warning("Unable to find ES data: {exc}".format(exc=exc))
```
Comment (bentsi) [inline sdcm/results_analyze/test.py:486]: use lazy interpolation
https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation

File: sdcm/results_analyze/test.py [hunk 2]
```diff
@@ -226,6 +226,7 @@ class SoftwareVersions(ClassBase):
     scylla_enterprise_server: ScyllaEnterpriseVersionInfo = None
 
     @property
+    # pylint: disable=unsubscriptable-object
```
Comment (bentsi) [inline sdcm/results_analyze/test.py:229]: why did you add it here? it shouldn't be subscriptable object

File: sdcm/results_analyze/test.py [hunk 3]
```diff
@@ -538,10 +544,15 @@ def get_same_tests_query(self):
...
-        es_result = ES().search(index=self._es_data['_index'], q=es_query, filter_path=filter_path,
-                                size=10000)  # pylint: disable=unexpected-keyword-arg
-        es_result = es_result.get('hits', {}).get('hits', None) if es_result else None
+        try:
+            es_query = self.get_same_tests_query()
+            es_result = ES().search(index=self._es_data['_index'], q=es_query, filter_path=filter_path,
+                                    size=10000)  # pylint: disable=unexpected-keyword-arg
+            es_result = es_result.get('hits', {}).get('hits', None) if es_result else None
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.warning("Unable to find ES data: {exc}".format(exc=exc))
```
Comment (bentsi) [inline sdcm/results_analyze/test.py:553]: use lazy interpolation

File: sdcm/results_analyze/test.py (hunk 1 above)
Comment (k0machi) [reply sdcm/results_analyze/test.py]: As we found out, outdated pylint version doesn't support union type hints, fixed and removed

File: sdcm/results_analyze/test.py (hunk 2 above)
Comment (k0machi) [reply sdcm/results_analyze/test.py]: pylint version I had didn't support proper linting for union type hints, fixed and removed that line

File: sdcm/results_analyze/test.py (hunk 3 above)
Comment (k0machi) [reply sdcm/results_analyze/test.py]: done

File: sdcm/results_analyze/test.py (hunk 1 above)
Comment (k0machi) [reply sdcm/results_analyze/test.py]: done

Comment (bentsi) [reply sdcm/results_analyze/test.py]: I don't see the change

File: sdcm/results_analyze/test.py (hunk 3 above)
Comment (bentsi) [reply sdcm/results_analyze/test.py]: same here, don't see the change

File: sdcm/results_analyze/test.py (hunk 1 above)
Comment (k0machi) [reply sdcm/results_analyze/test.py]: corrected for real this time, somehow missed both

File: sdcm/results_analyze/test.py (hunk 3 above)
Comment (k0machi) [reply sdcm/results_analyze/test.py]: corrected for real this time, somehow missed both

**Review by bentsi — APPROVED** [review]
